api/secretsmanager/client.go

```diff
@@ -3,452 +3,464 @@ package secretsmanager
-	var object struct {
-		ID string `json:"id"`
-	}
+// CreateScriptTemplate create script template.
+func (c *SecretsManager) CreateScriptTemplate(template *ScriptTemplate) (response.Identifier, error) {
+	identifier := response.Identifier{}
 
-	_, err := s.api.
+	_, err := c.api.
 		URL("/secrets-manager/api/v1/script-template").
-		Post(&t, &object)
+		Post(&template, &identifier)
 
-	return object.ID, err
+	return identifier, err
 }
 
-// ScriptTemplate get script template by id
-func (s *Client) ScriptTemplate(templateId string) (*ScriptTemplate, error) {
+// GetScriptTemplate get script template by id.
+func (c *SecretsManager) GetScriptTemplate(templateID string) (*ScriptTemplate, error) {
 	p := &ScriptTemplate{}
 
-	_, err := s.api.
-		URL("/secrets-manager/api/v1/script-template/%s", url.PathEscape(templateId)).
+	_, err := c.api.
+		URL("/secrets-manager/api/v1/script-template/%s", templateID).
 		Get(&p)
 
 	return p, err
 }
 
-// UpdateScriptTemplate update existing script template
-func (s *Client) UpdateScriptTemplate(templateId string, t ScriptTemplate) error {
-	_, err := s.api.
-		URL("/secrets-manager/api/v1/script-template/%s", url.PathEscape(templateId)).
-		Put(t)
+// UpdateScriptTemplate update script template.
+func (c *SecretsManager) UpdateScriptTemplate(templateID string, template *ScriptTemplate) error {
+	_, err := c.api.
+		URL("/secrets-manager/api/v1/script-template/%s", templateID).
+		Put(&template)
 
 	return err
 }
 
-// DeleteScriptTemplate delete a script template
-func (s *Client) DeleteScriptTemplate(templateId string) error {
-	_, err := s.api.
-		URL("/secrets-manager/api/v1/password-policy/%s", url.PathEscape(templateId)).
+// DeleteScriptTemplate delete script template.
+func (c *SecretsManager) DeleteScriptTemplate(templateID string) error {
+	_, err := c.api.
+		URL("/secrets-manager/api/v1/password-policy/%s", templateID).
 		Delete()
 
 	return err
 }
 
-// CompileScript compile script with test data
-func (s *Client) CompileScript(r CompileScriptRequest) (string, error) {
-	var object struct {
-		Script string `json:"script"`
-	}
+// CompileScript compile script with test data.
+func (c *SecretsManager) CompileScript(compile CompileScript) (CompileScriptResponse, error) {
+	compiled := CompileScriptResponse{}
 
-	_, err := s.api.
+	_, err := c.api.
 		URL("/secrets-manager/api/v1/script-template/compile").
-		Post(&r, &object)
+		Post(&compile, &compiled)
 
-	return object.Script, err
+	return compiled, err
 }
 
-// MARK: Target domains
-// TargetDomains lists all target domains
-func (s *Client) TargetDomains(offset, limit int, sortkey, sortdir string) (TdResult, error) {
-	result := TdResult{}
-	filters := Params{
-		Offset:  offset,
-		Limit:   limit,
-		Sortkey: sortkey,
-		Sortdir: sortdir,
+// MARK: Manage Secrets
+// GetHostSecretMetadata get host secret metadata for all accounts.
+func (c *SecretsManager) GetHostSecretMetadata(hostID string) (*PostHostSecret, error) {
```

Comment:
PostHostSecret sounds strange as a type

Reply:
Yep it does, this is taken from the original struct definition. Same goes also for the create handler which takes PostHostSecret and returns a PostHostSecret...
Let me remove the "Post" suffix for this struct, this should make it a bit less confusing.